dace/codegen/targets/fpga.py

```diff
@@ -1021,10 +1022,14 @@ def _generate_MapEntry(self, sdfg, dfg, state_id, node, function_stream,
             # Generate nested loops
             if not isinstance(node, dace.sdfg.nodes.PipelineEntry):
 
-                if is_innermost and not fully_degenerate:
-                    self.generate_flatten_loop_pre(result, sdfg, state_id, node)
+
 
                 for i, r in enumerate(node.map.range):
+
+                    if is_innermost and not fully_degenerate and not is_degenerate[i]:
```

Comment:
Now it will put one at every loop, right? Is that we want?

Reply:
yes, one per loop. I don't see how to generate it differently